rs/xnet/payload_builder/src/lib.rs

```diff
@@ -55,6 +56,7 @@ use std::{
     sync::{Arc, Mutex},
```

Comment:
Not directly related to this change: would it help to swap this for some async-aware mutex?

Reply:
The async mutexes are usually slower than the sync variant. The only reason why we should swap it if we lock the mutex across await points. Which I don't think it is the case otherwise clippy would have complained.